src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -851,30 +854,42 @@ internal SSHConnection[] ParseSSHConnectionHashTable()
                         throw new PSArgumentException(RemotingErrorIdStrings.InvalidSSHConnectionParameter);
                     }
 
-                    string paramValue = item[paramName] as string;
-                    if (string.IsNullOrEmpty(paramValue))
+                    if (paramName.Equals(PortParameter, StringComparison.OrdinalIgnoreCase))
```

Comment:
I agree with @iSazonov, it would be nice to keep the if-else-if blocks.  You can create a helper method to get a string or int parameter and throw exception a needed.
```
private static string GetStringParameter(object param)
{
    var paramValue = param as string;
    if (string.IsNullOrEmpty(paramValue)) { throw new PSArgumentException("..."); }
    return paramValue;
}

private static int GetIntParameter(object param)
{
    int? paramValue = param as int?;
    if (paramValue == null) { throw new PSArgumentException("..."); }
    return paramValue.Value;
}

if (paramName.Equals(PortParameter, StringComparison.OridinalIgnoreCase))
{
    connectionInfo.Port = GetIntParameter(item[paramName]);
}
else if ...
```
We don't need the Convert.ToInt32() and I believe the integer is only un-boxed once.